retrieval/target.go

```diff
@@ -451,33 +338,23 @@ func (t *Target) scrape(appender storage.SampleAppender) (err error) {
 	sdec := expfmt.SampleDecoder{
 		Dec: dec,
 		Opts: &expfmt.DecodeOptions{
-			Timestamp: model.TimeFromUnixNano(start.UnixNano()),
+			Timestamp: model.TimeFromUnixNano(time.Now().UnixNano()),
```

Comment:
Previously the scrape timestamp was always the very start of the scrape, which we assumed to be the most accurate guess for most use cases (the metrics are all snapshotted at the beginning of the request, then transferred). Any particular reason why this changes the timestamp to the beginning of the body reception?